pkg/cloudevents/eventcontext_v01.go

```diff
@@ -76,7 +76,21 @@ func (ec *EventContextV01) SetExtension(name string, value interface{}) error {
 
 // Clone implements EventContextConverter.Clone
 func (ec EventContextV01) Clone() EventContext {
-	return ec.AsV01()
+	ev01 := ec.AsV01()
+	ev01.deepCloneExtensions()
```

Comment:
normally a clone method returns the clone

Reply:
So do you think we need another method, like `DeepClone()`, for this?

Reply:
It is very strange to have a method that means "you know the stuff that is inside you... yah just make a copy of yourself and save it internally."

Reply:
ah sorry i didn't catch up you were talking about `deepCloneExtensions()`! Yeah, you're right, let me fix it